### PR TITLE
Drive merge process from tuples not rows and fix merge table suppression

### PR DESCRIPTION
**Pull request by robmen**
And other code clean up

### CODE REVIEW

File: src/WixToolset.Core.WindowsInstaller/Bind/BindDatabaseCommand.cs [hunk 1]
```diff
@@ -412,6 +403,13 @@ public IBindResult Execute()
 
             this.ValidateComponentGuids(output);
 
+            // We can create instance transforms since Component Guids and Outputs are created.
+            if (output.Type == OutputType.Product)
+            {
+                var command = new CreateInstanceTransformsCommand(section, output, tableDefinitions, this.BackendHelper);
+                command.Execute();
+            }
+
```
Comment (rseanhall) [inline src/WixToolset.Core.WindowsInstaller/Bind/BindDatabaseCommand.cs:412]: Shouldn't this be after the following check if any errors occurred?

File: src/WixToolset.Core.WindowsInstaller/Bind/AddBackSuppresedSequenceTablesCommand.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    using System.Collections.Generic;
+    using WixToolset.Data.Tuples;
+    using WixToolset.Data.WindowsInstaller;
+
+    /// <summary>
+    /// Add back possibly suppressed sequence tables since all sequence tables must be present
+    /// for the merge process to work. We'll drop the suppressed sequence tables again as
+    /// necessary.
+    /// </summary>
+    internal class AddBackSuppresedSequenceTablesCommand
```
Comment (rseanhall) [inline src/WixToolset.Core.WindowsInstaller/Bind/AddBackSuppresedSequenceTablesCommand.cs:15]: AddBackSuppresed -> AddBackSuppressed

Comment (robmen) [reply src/WixToolset.Core.WindowsInstaller/Bind/AddBackSuppresedSequenceTablesCommand.cs:15]: Ugh. Always typo that even though I know there are too many 's's in it. :)

File: src/WixToolset.Core.WindowsInstaller/Bind/BindDatabaseCommand.cs (hunk 1 above)
Comment (robmen) [reply src/WixToolset.Core.WindowsInstaller/Bind/BindDatabaseCommand.cs:412]: I suppose. Still looking at moving cabinet build and that is my focus.

**Pull request merged**